Revert "adding check_dt to fix issue #1 for mismatching columns format"

matbmeijer opened the pull request:
Reverts matbmeijer/JirAgileR#3 as package did not work with fixes and only works in specific non-universal cases

matbmeijer committed:
Revert "adding check_dt to fix issue #1 for mismatching columns format"

Dschaykib commented:
can you explain to me a bit more, why it did not work? Which non-universal cases did I cover?

matbmeijer commented:
Hi @Dschaykib,

Sorry for the late reply. First of all, thank you very much for getting involved :thumbsup:. Your interest is the reason I am investing much more time in this package than I expected. If I remember correctly, I had problems obtaining the fields `status, priority, created, reporter, summary, description, assignee, updated, issuetype, fixVersions` from a project. To be honest, I did not dig to deep into why it failed -  I did not like how I had structured the previous steps of the`JiraJSON2df()` function, and wanted to restructure/clean up the package for multiple reasons:
1. My code was not especially robust - I wrote it for a single use case - and it contained multiple things that should preferably not be in a package (e.g. `sapply()`, for loops etc.). Additionally, it dealt with multiple JIRA fields on a luck basis (if not this and not this, unlist and concatonate). It also contained functions such as recursive `unlist()` functions, which mess up the original classes (changing dates and numbers to characters, for example).
2. I was actually not returning all the data from the API response, but only a portion of it. I prefer to have a package, which is opinionated in structure (e.g. flattened data.frame, tidy data, etc.), but not in the content I return from the original API. That's why now the package returns much more details of each field.
2. The code was difficult to debug. I did not like the multiple if clauses. Instead I changed it now to a switch, with a dedicated function to each field, which makes debugging and further development much easier.
3. I did not want to increase my dependency on the `data.table` package. Don't misunderstand me, the `data.table` package is fantastic - I use it for my personal analysis - but I try to keep everything as much in base as possible so the number of package dependencies keeps being low so I am not affected by package changes (I already have to deal with enough API changes  :man_facepalming:). That's also why I have removed the `rbindlist(fill=TRUE)` function. In other words, I both like `data.table` and the whole `tidyverse`, but I try to stay away from it for package development (the purrr package would have been fantastic for the whole nested lists here). 

I hope that responds your question - I would be thankful for any kind of feedback on the package. Thank you!

Dschaykib commented:
No problem, thanks for the explanation! I can relate to trying to develop with as little dependencies as possible - even though I am a fan of data.table ;)
I will use the package for one of my projekts and let you know if I find anything else.
